File: lab4/lang-tests/test-4/scope.scala
object Scope {
    
    var y : Int = 0;
    def fun_print(z: Int) {
        println("El valor de z en fun_print es: " + z);
        println("El valor de y en fun_print es: " + y);
        println("Java tiene alcance Estático");
    }

    def rewrite(z: Int){
        var y : Int = 3;
        println("El valor de y en rewrite es: " + y);
        return fun_print(y + z);
    }

    def main(args: Array[String]) {

        // val a = new Alcance()
        var y : Int = 2
        println(" la variable y en main vale: " + y);
        rewrite(y)
    }
}

//scala es de alcance estatico

File: lab4/lang-tests/test-4/scope.java
class Scope {
    
    int y = 1;
    int fun_print(int z) {
        System.out.format("El valor de y es: %d\n", z);
        System.out.format("El valor de y es: %d\n", y);
        System.out.format("Java tiene alcance Estático\n");
        return z;
    }

    int fun_write(int z){
        int y = 3;
        System.out.format("El valor de y en fun_write es: %d\n", y);
        return fun_print(y + z);
    }

    public static void main (String args[]) {
        
        int y = 2;
        System.out.format("El valor de y en main es: %d\n", y);
        Scope a = new Scope();
        a.fun_write(y);
  }
}
